ui/src/app/applications/components/application-sync-panel/application-sync-panel.scss

```diff
@@ -2,11 +2,30 @@
     margin-top: 0.5em;
     white-space: nowrap;
 
-    label {
-        max-width: 38em;
-        overflow: hidden;
-        text-overflow: ellipsis;
+    .container {
+        max-width: 32em;
+        white-space: nowrap;
         display: inline-block;
-        vertical-align: middle;
+        margin-right: 0.3em;
+
+        label::before,
+        label::after {
+            vertical-align: middle;
+            display: inline-block;
+            max-width: 50%;
+            overflow: hidden;
+            white-space: pre;
+        }
+
+        label::before {
+            content: attr(content-start);
+            text-overflow: ellipsis;
+        }
+
+        label::after {
+            content: attr(content-end);
+            text-overflow: clip;
+            direction: rtl;
```

Comment:
Adding a comment here after `direction: rtl;` as `// This is for ellipse in the middle`.